testbeam_analysis/track_analysis.py

```diff
@@ -1220,8 +1225,9 @@ def _fit_tracks_kalman_loop(track_hits, dut_fit_selection, pixel_size, n_pixels,
     sel = dut_selection[:-1]
     z_diff = z_positions[sel + 1] - z_positions[sel]
 
-    if additional_scatter is True:  # need to shift dut fit selection in case of additional scattering plane
-        dut_fit_selection[np.where(dut_fit_selection > (index_scatter - 1))[0][0]:] = dut_fit_selection[np.where(dut_fit_selection > (index_scatter - 1))[0][0]:] + 1
+    if add_scattering_plane:
+        for i in range(len(index_scatter)):  # need to shift dut fit selection in case of additional scattering plane
+                dut_fit_selection[np.where(dut_fit_selection > (index_scatter[i] - 1))[0][0]:] = dut_fit_selection[np.where(dut_fit_selection > (index_scatter[i] - 1))[0][0]:] + 1
```

Comment:
This has also to be changed, since the dut_index is different if additional scattering planes are considered. E.g. if we want to use DUT 0 and 1 in fit, and add a scattering plane in between, the dut_indices are now: 0, 1 (scattering plane) and 2. Then `dut_fit_selection` has to be changed to  `dut_fit_selection = [0, 2] `.